### DISCUSSION TITLE
Expose drawer partially for opening and closing button.

### DISCUSSION THREAD

**Discussion by EladKarni** [saadeghi/daisyui#2982] [Q&A] [answered]
I'm trying to create a layout that has a drawer, but exposes part of it so it can be clicked to open it. Any way of doing this with the existing drawer component or is this something I'd need to make from scratch? I attached a video as an example from the Material UI library.

https://daisyui.com/components/drawer/

Here is what I'm trying to do:

https://github.com/saadeghi/daisyui/assets/39969481/98b37fac-c2cf-4d1f-be19-d23342bf3aab

**Comment by saadeghi** [comment, marked as answer]
daisyUI `drawer` is a grid layout with sidebar and content that can collapse the sidebar if needed.  
In this screenshot, it's a different layout. Some people call it a `bottom sheet` and it's a common design pattern in iOS. We don't have this in daisyUI currently, but I just made a quick example for you:  
https://play.tailwindcss.com/NBg7Q0zzWp

I used a checkbox to open/close (CSS-only) but you can also use JS and apply class names conditionally to open or close it.

Let me know if you have a question.
